internal/globaldefault/default.go

```diff
@@ -53,17 +53,20 @@ func Prepare(cfg DefaultConfigurer, subshell subshell.SubShell) error {
 		"PATH": binDir,
 	}
 
-	if err := subshell.WriteUserEnv(cfg, envUpdates, sscommon.DefaultID, true); err != nil {
+	// Configure available shells
+	shell := subshell.New(cfg)
```

Comment:
This was receiving it as an argument before, not constructing it here.

The point of the change I requested is that we're not manually constructing subshell every time and instead pass it around.